src/libraries/System.Security.Cryptography.X509Certificates/tests/X509StoreMutableTests.Android.cs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Security.Cryptography.X509Certificates.Tests
+{
+    [PlatformSpecific(TestPlatforms.Android)]
+    public static partial class X509StoreMutableTests
+    {
+        public static bool PermissionsAllowStoreWrite { get; } = true;
+
+        [Fact]
+        public static void AddRemove_Cert()
+        {
+            using (var store = new X509Store(StoreName.My, StoreLocation.CurrentUser))
+            using (var cert = new X509Certificate2(TestData.MsCertificate))
+            {
+                store.Open(OpenFlags.ReadWrite);
+
+                // Make sure cert is not already in the store
+                store.Remove(cert);
+                Assert.False(IsCertInStore(cert, store), "Certificate should not be found on pre-condition");
+
+                // Add
+                store.Add(cert);
+                Assert.True(IsCertInStore(cert, store), "Certificate should be found after add");
+
+                // Remove
+                store.Remove(cert);
+                Assert.False(IsCertInStore(cert, store), "Certificate should not be found after remove");
+            }
+        }
+
+        [Theory]
+        [InlineData(nameof(TestData.RsaCertificate), TestData.RsaCertificate, TestData.RsaPkcs8Key)]
+        [InlineData(nameof(TestData.EcDhCertificate), TestData.EcDhCertificate, TestData.EcDhPkcs8Key)]
+        [InlineData(nameof(TestData.ECDsaCertificate), TestData.ECDsaCertificate, TestData.ECDsaPkcs8Key)]
+        public static void AddRemove_CertWithPrivateKey(string testCase, string certPem, string keyPem)
+        {
+            _ = testCase;
+            using (var store = new X509Store(StoreName.My, StoreLocation.CurrentUser))
+            using (var cert = X509Certificate2.CreateFromPem(certPem, keyPem))
+            {
+                store.Open(OpenFlags.ReadWrite);
+
+                // Make sure cert is not already in the store
+                store.Remove(cert);
+                Assert.False(IsCertInStore(cert, store), "Certificate should not be found on pre-condition");
+
+                // Add
+                store.Add(cert);
+                Assert.True(IsCertInStore(cert, store), "Certificate should be found after add");
+                Assert.True(StoreHasPrivateKey(store, cert), "Certificate in store should have a private key");
```

Comment:
Is there also a test that adds a cert as a plain cert, then adds it with a private key, to see that it correctly joins them?  (Perhaps it's in the other partial file already?)

Reply:
I believe that is `AddPrivateAfterPublic` in the other partial file.